CanvasColorSpaceProposal.md

```diff
@@ -138,21 +138,35 @@ ImageBitmap objects are augmented to have an internal color space attribute of t
 
 IDL
 <pre>
+
+enum ImageDataStorageFormat {
+  "uint8", // default
+  "uint16",
+  "float32",
+};
+
 typedef (Uint8ClampedArray or Uint16ClampedArray or Float32Array) ImageDataArray;
 
-[Constructor(unsigned long sw, unsigned long sh, optional CanvasColorSpace colorSpace = "legacy-srgb"),
- Constructor(ImageDataArray data, unsigned long sw, optional unsigned long sh, optional CanvasColorSpace colorSpace),
+dictionary ImageDataColorSettings {
+  CanvasColorSpace colorSpace = "srgb";
+  ImageDataStorageFormat storageFormat = "uint8";
+};
+
+[Constructor(unsigned long sw, unsigned long sh, optional ImageDataColorSettings imageDataColorSettings),
+ Constructor(ImageDataArray data, unsigned long sw, optional unsigned long sh, optional ImageDataColorSettings imageDataColorSettings),
  Exposed=(Window,Worker)]
 interface ImageData {
   readonly attribute unsigned long width;
   readonly attribute unsigned long height;
   readonly attribute ImageDataArray data;
-  readonly attribute CanvasColorSpace colorSpace;
+  readonly attribute ImageDataColorSettings colorAttributes;
 };
 </pre>
 
-* getImageData() produces an ImageData object in the same color space as the source canvas, using an ImageDataArray of a type that is appropriate for the specified pixelFormat (smallest possible numeric size that guarantees no loss of precision)
+* When using the constructor with ImageDataArray parameter, the array type must respect ImageDataColorSettings.ImageDataStorageFormat. Otherwise, the return object will be undefined.
+* getImageData() produces an ImageData object with the same color space as the source canvas, using an ImageDataArray of a type that is appropriate for the pixelFormat of the source canvas (smallest possible numeric size that guarantees no loss of precision).
 * putImageData() performs a color space conversion to the color space of the destination canvas.
+* The proper ImageDataStorageFormat for "10-10-10-2" and "12-12-12-12" canvas pixel formats is "uint16". getImageData() and putImageData() use bit shifting for numerical scaling.
```

Comment:
Actually, the correct conversion Is more complex that just shifting if you want to get the closest value in the new space. For example 10 to 16 bit conversion is: (x << 6) + (x >> 4).  Besides, there is no need to document the conversion algorithm here.